web3/exceptions.py

```diff
@@ -259,7 +259,7 @@ class ContractLogicError(Web3Exception):
     Raised on a contract revert error
     """
 
-    def __init__(self, message: Optional[str] = None, data: Optional[str] = None):
+    def __init__(self, message: Optional[str] = None, data: Optional[Any] = None):
```

Comment:
Also, some type narrowing might be more appropriate. Maybe `Union[str, Dict[str, str]]`.

Reply:
Thanks for pointing to that dict instance, that's what I was looking for. Opening it up to Any is technically backwards compatible 😄 though it'd be great to lock this down to str | dict if possible.

Reply:
It shouldn't be breaking if it goes from `str` -> `Any`. If it was `Any` -> `str` or something narrowing, then it would be breaking. But I think a Union would be fine too.

Reply:
Sounds good, I'll narrow it down 👌🏼